java-squid/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -505,29 +504,48 @@ public MethodJavaSymbol overriddenSymbol() {
         return null;
       }
       TypeJavaSymbol enclosingClass = enclosingClass();
+      boolean unknownFound = false;
       for (JavaType.ClassJavaType superType : enclosingClass.superTypes()) {
         MethodJavaSymbol overridden = overriddenSymbolFrom(superType);
         if (overridden != null) {
-          return overridden;
+          if (!overridden.isUnknown()) {
+            return overridden;
+          } else {
+            unknownFound = true;
+          }
         }
       }
+      if (unknownFound) {
+        return Symbols.unknownMethodSymbol;
+      }
       return null;
     }
 
     @Nullable
     private MethodJavaSymbol overriddenSymbolFrom(JavaType.ClassJavaType classType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "overriddenSymbolFrom" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)